src/code_translation/NodeSelector.py

```diff
@@ -28,12 +28,12 @@ class NodeSelector(cst.CSTVisitor):
     def __init__(self) -> None:
 
         self.variables: dict[str, Node] = {}
-        self.interesting_nodes: dict[cst.CSTNode, Node] = {}
+        self.interesting_nodes: dict[cst.Assign, Node] = {}
```

Comment:
this is actually quite an interesting and non-trivial change (because it also affects the NodeReplacer) but can you think of any nodes except Assignments which we will replace? I would just keep that for now and look if we need to generalize some things

Reply:
Inplace operations would be Calls but I'm not sure if this is in-scope or not. It should get passed to our IR, but we'd need an option to remove them afterwards.